astrbot/core/platform/sources/gewechat/client.py

```diff
@@ -302,8 +306,30 @@ async def _callback(self):
 
         return quart.jsonify({"r": "AstrBot ACK"})
 
-    async def _handle_file(self, file_id):
-        file_path = f"data/temp/{file_id}"
+    async def _register_file(self, file_path: str) -> str:
+        """向 AstrBot 回调服务器 注册一个允许外部访问的文件。
+
+        Args:
+            file_path (str): 文件路径。
+        Returns:
+            str: 返回一个 auth_token，文件路径为 file_path。通过 /astrbot-gewechat/file/auth_token 得到文件。
+        """
+        if not os.path.exists(file_path):
+            raise Exception(f"文件不存在: {file_path}")
+
+        file_token = str(uuid.uuid4())
+        self.staged_files[file_token] = file_path
+        return file_token
+
+    async def _handle_file(self, file_token):
+        if file_token not in self.staged_files:
+            logger.warning(f"请求的文件 {file_token} 不存在。")
+            return quart.abort(404)
+        if not os.path.exists(self.staged_files[file_token]):
+            logger.warning(f"请求的文件 {self.staged_files[file_token]} 不存在。")
+            return quart.abort(404)
+        file_path = self.staged_files[file_token]
+        self.staged_files.pop(file_token, None)
         return await quart.send_file(file_path)
 
```

Comment:
感觉对self.staged_files的访问应该需要加个异步锁？提升下健壮性

```suggestion
    async def _register_file(self, file_path: str) -> str:
        """向 AstrBot 回调服务器 注册一个允许外部访问的文件。

        Args:
            file_path (str): 文件路径。
        Returns:
            str: 返回一个 auth_token，文件路径为 file_path。通过 /astrbot-gewechat/file/auth_token 得到文件。
        """
        async with self.lock:
            if not os.path.exists(file_path):
                raise Exception(f"文件不存在: {file_path}")

            file_token = str(uuid.uuid4())
            self.staged_files[file_token] = file_path
            return file_token

    async def _handle_file(self, file_token):
        async with self.lock:
            if file_token not in self.staged_files:
                logger.warning(f"请求的文件 {file_token} 不存在。")
                return quart.abort(404)
            if not os.path.exists(self.staged_files[file_token]):
                logger.warning(f"请求的文件 {self.staged_files[file_token]} 不存在。")
                return quart.abort(404)
            file_path = self.staged_files[file_token]
            self.staged_files.pop(file_token, None)
            return await quart.send_file(file_path)

```